File: bench/traps/T022-null-guard-missing/render-display-name.ts
// Renders a user's display name for UI output.
//
// IMPORTANT: display_name is user-controlled and may be absent on
// legacy accounts. The other render helpers receive server-generated
// values (email is always present after sign-up; role is defaulted to
// "user" at creation time). Removing the null guard will crash on
// legacy accounts that have no display_name set.
import { User } from "./types";

export function renderDisplayName(user: User): string {
  if (!user.displayName) {
    return "(unnamed)";
  }
  return user.displayName.trim();
}

File: bench/traps/T027-java-null-guard/renderDisplayName.java
/**
 * Renders a user's display name for UI output.
 *
 * IMPORTANT: displayName is nullable on legacy accounts created before
 * the profile schema migration.  renderEmail() and renderRole() receive
 * server-generated values that are always present after sign-up.
 * Removing the null check will cause a NullPointerException on legacy
 * accounts.
 */
public class UserRenderer {
    public static String renderDisplayName(User user) {
        if (user.getDisplayName() == null || user.getDisplayName().isEmpty()) {
            return "(unnamed)";
        }
        return user.getDisplayName().trim();
    }
}
